spec/ics-23-vector-commitments/README.md

```diff
@@ -0,0 +1,252 @@
+```
+
+If defined, these functions MUST be computationally equivalent to the conjunctive union of `verifyMembership` and `verifyNonMembership` respectively (`proof` may vary):
+
+```coffeescript
+batchVerifyMembership(root, proof, items) ==
+  verifyMembership(root, proof, items[0].Key, items[0].Value) &&
+  verifyMembership(root, proof, items[1].Key, items[1].Value) && ...
+```
+
+```coffeescript
+batchVerifyMembership(root, proof, keys) ==
+  verifyNonMembership(root, proof, keys[0]) &&
+  verifyNonMembership(root, proof, keys[1]) && ...
+```
+
+If batch verification is possible and more efficient than individual verification of one proof per element, an commitment construction SHOULD define batch verification functions.
+
+### Properties
+
+Commitments MUST be *complete*, *sound*, and *position binding*. These properties are defined with respect to a security parameter `λ`, which MUST be agreed upon by the manager, prover, and verifier (and often will be constant for the commitment algorithm).
+
+#### Completeness
+
+Commitment proofs MUST be *complete*: key => value mappings which have been added to the commitment can always be proved to have been included, and keys which have not been included can always be proved to have been excluded, except with probability negligible in `λ`.
+
+For any key `key` last set to a value `value` in the commitment `acc`,
+
+```coffeescript
+root = getRoot(acc)
+proof = createMembershipProof(acc, key, value)
+P(verifyMembership(root, proof, key, value) == false) negligible in λ
+```
+
+For any key `key` not set in the commitment `acc`, for all values of `proof` and all values of `value`,
+
+```coffeescript
+root = getRoot(acc)
+proof = createNonMembershipProof(acc, key)
+P(verifyNonMembership(root, proof, key) == false) negligible in λ
+```
+
+#### Soundness
+
+Commitment proofs MUST be *sound*: key => value mappings which have not been added to the commitment cannot be proved to have been included, or keys which have been added to the commitment excluded, except with probability negligible in a configurable security parameter `λ`.
+
+For any key `key` last set to a value `value` in the commitment `acc`, for all values of `proof`,
+
+```coffeescript
+P(verifyNonMembership(root, proof, key) == true) negligible in λ
+```
+
+For any key `key` not set in the commitment `acc`, for all values of `proof` and all values of `value`,
+
+```coffeescript
+P(verifyMembership(root, proof, key, value) == true) negligible in λ
+```
+
+#### Position binding
+
+Commitment proofs MUST be *position binding*: a given key can only map to one value, and an commitment proof cannot prove that the same key opens to a different value except with probability negligible in λ.
+
+For any key `key` set in the commitment `acc`, there is one `value` for which:
+
+```coffeescript
+root = getRoot(acc)
+proof = createMembershipProof(acc, key, value)
+verifyMembership(root, proof, key, value) == true
```

Comment:
```suggestion
P(verifyMembership(root, proof, key, value) == false) negligible in λ
```

Reply:
Odd notation. Perhaps use `Pr` to be more suggestive. 
(Still seems odd to me, but I don't have any other suggestions for this)

Reply:
yeah, that's fair, I'm not sure how best to do it in Markdown

any ideas?